### PR TITLE
funding.json

### PR DESCRIPTION
**Pull request by jemgillam**
## Description

I have begun a `funding.json` file. I have kept the comments in it for now, but they will need removing before the final merge into the codebase. Some well-known links will need to be created across our websites. Moved from https://github.com/graphile/crystal/pull/2268

~I chose Graphile Worker and PostGraphile as our projects, we can add more, but I feel Grafast might be too new. The whole thing links to Graphile as an organisation any way.~

I put in our plans from $100/mo and above.

See [floss.fund/funding-manifest](https://floss.fund/funding-manifest/)

This is a new initiative from [zerodha.tech](https://zerodha.tech/)
They launched it last month, the first funding cycle from them is going to start in the new year.

If this initiative gets momentum, then it is worth having a funding.json file anyway, even if FLOSS/fund don't grant us any sponsorship.

Here is a directory of others who have already set this up: [dir.floss.fund/browse/projects](https://dir.floss.fund/browse/projects)

Here are the tags we can choose from for our projects: [floss.fund/static/project-tags.txt](https://floss.fund/static/project-tags.txt)

## To Do 

- [x] well-known URIs for 
  - [x] worker.graphile.org
  - [x] build.graphile.org 
  - [x] star.graphile.org
  - [x] postgraphile.org
  - [x] grafast.org
  - [x] ~~graphile.org~~ Not needed
- [x] And in the repos too?
  - [x] worker
  - [x] utils
  - [x] build
  - [x] postgraphile
  - [x] grafast
  - [x] graserv
  - [x] migrate
  - [x] ruru 	 
- [x] include all projects over x stars / downloads
  - [x] grafast
  - [x] grafserv
  - [x] graphile-build
  - [x] graphile-migrate
  - [x] graphile-worker
  - [x] graphile-utils
  - [x] postgraphile
  - [x] ruru 
- [x] remove comments
- [ ] review wording in descriptions
- [x] readme for star.graphile.org - `crystal/utils` repo

### CODE REVIEW

**Comment by jemgillam** [comment]
I have added the funding-manifest files to various of our websites

@Benjie am I right in thinking the GitHub repos need them too?

**Comment by jemgillam** [comment]
To add to the above, vue.js have one here https://github.com/vuejs/core/blob/main/.well-known/funding-manifest-urls

I have no idea if this works as intended, but it is shown on the FOSS/fund directory as so: https://dir.floss.fund/view/project/@vuejs.org/vuejs

**Comment by benjie** [comment]
Good research; go with that... I guess?

**Comment by jemgillam** [comment]
To add onto links for the repos, I have decided to link to  
```
https://github.com/graphile/crystal/tree/main/graphile-build/graphile-build/.well-known/funding-manifest-urls
```
rather than  
```
https://raw.githubusercontent.com/graphile/crystal/tree/main/graphile-build/graphile-build/.well-known/funding-manifest-urls
```
even though the second link is purely the file, the domain is different and so isn't exactly what is required. This approach seems to work for other projects :shrug:

**Comment by jemgillam** [comment]
@benjie ready to look over. I am adding a readme to `crystal/utils` for the "star" collection of tools/utilities sometime today.

I think we want to discuss some of the wording for the descriptions

File: funding.json [hunk 1]
```diff
@@ -0,0 +1,243 @@
+{
+  "version": "v1.0.0",
+  "entity": {
+    "type": "organisation",
+    "role": "owner",
+    "name": "Graphile",
+    "email": "fundingjson@graphile.com",
+    "description": "At Graphile we love GraphQL so much we named ourselves for our love of it!",
```
Comment (benjie) [inline funding.json:8]:
```suggestion
    "description": "Devtools for Node.js, PostgreSQL and GraphQL focussed on development and execution efficiency.",
```

File: funding.json [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+          "development",
+          "programming",
+          "software-engineering",
+          "web-development"
+        ]
+      },
+      {
+        "guid": "graphile-star",
+        "name": "Graphile Star",
+        "description": "A group of useful Graphile utilities, including graphile-config which provides a standard plugin interface and helpers which can be used across the entire Graphile suite; graphile-export which exports a GraphQL Schema (or other code) as executable JavaScript; tamedevil a module to help tame eval; and pg-sql2 and pg-introspection.",
```
Comment (benjie) [inline funding.json:100]:
```suggestion
        "description": "A collection of utility modules including: graphile-config (standardized plugins and presets infrastructure); graphile-export (export a dynamic GraphQL Schema as executable JavaScript); tamedevil (makes dealing with `eval` safer); pg-sql2 (SQLi-resistant and speedy SQL templating in Node.js) and pg-introspection.",
```

File: funding.json [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+          "url": "https://github.com/graphile/crystal/tree/main/utils",
+          "wellKnown": "https://github.com/graphile/crystal/tree/main/utils/.well-known/funding-manifest-urls"
+        },
+        "licenses": ["spdx:MIT"],
+        "tags": ["developer-tools", "development"]
+      },
+      {
+        "guid": "graphile-worker",
+        "name": "Graphile Worker",
+        "description": "Job queue for PostgreSQL running on Node.js - allows you to run jobs (e.g. sending emails, performing calculations, generating PDFs, etc) in the background so that your HTTP response/application code is not held up. Can be used with any PostgreSQL-backed application.",
```
Comment (benjie) [inline funding.json:115]:
```suggestion
        "description": "Node.js job queue powered by PostgreSQL - schedule jobs (sending emails, converting images, generating PDFs, etc) in the background so your website and API requests don't have to wait.",
```

**Review by benjie — CHANGES_REQUESTED** [review]

**Comment by jemgillam** [comment]
https://dir.floss.fund/submit gives an error, so a `well-known` URI is also needed on graphile.org? I don't know if that makes sense from their descriptions... 

```
entity.webpageUrl.url and manifest URL host and paths do not match. Expected entity.webpageUrl.wellKnown for provenance check at https://graphile.org/*/.well-known/funding-manifest-urls
```

**Pull request merged**